src/ununpack/agent/checksum.c

```diff
@@ -126,22 +126,30 @@ int	CountDigits	(uint64_t Num)
  */
 Cksum *	SumComputeFile	(FILE *Fin)
 {
-  int rc;
-  SHA1Context sha1;
-  MyMD5_CTX md5;
+  gcry_md_hd_t checksumhandler;
+  gcry_error_t checksumError = 0;
   char Buffer[64];
   Cksum *Sum;
+  unsigned char *tempBuff;
   int ReadLen;
   uint64_t ReadTotalLen=0;
 
   Sum = (Cksum *)calloc(1,sizeof(Cksum));
   if (!Sum) return(NULL);
 
-  MyMD5_Init(&md5);
-  rc = SHA1Reset(&sha1);
-  if (rc)
+  checksumError = gcry_md_open(&checksumhandler, GCRY_MD_NONE, 0);
+  if (! checksumhandler)
+  {
+    LOG_ERROR("Unable to initialize checksum\n");
+    free(Sum);
+    return(NULL);
+  }
+  checksumError = gcry_md_enable(checksumhandler, GCRY_MD_MD5);
```

Comment:
check for the error code here as well 
 if (gcry_err_code(checksumError) != GPG_ERR_NO_ERROR)

Reply:
Done, please check.